modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/info/display/field/DDMFormValuesInfoFieldSetProvider.java

```diff
@@ -0,0 +1,243 @@
+			t, ddmFormFieldValue
+		).map(
+			infoFieldValues::add
+		);
+	}
+
+	private void _addNestedFields(
+		T t, DDMFormFieldValue ddmFormFieldValue,
+		List<InfoFieldValue> infoFieldValues) {
+
+		Map<String, List<DDMFormFieldValue>> nestedDDMFormFieldValuesMap =
+			ddmFormFieldValue.getNestedDDMFormFieldValuesMap();
+
+		for (Map.Entry<String, List<DDMFormFieldValue>> entry :
+				nestedDDMFormFieldValuesMap.entrySet()) {
+
+			List<DDMFormFieldValue> ddmFormFieldValues = entry.getValue();
+
+			ddmFormFieldValues.forEach(
+				nestedDDMFormFieldValue -> _addDDMFormFieldValue(
+					t, nestedDDMFormFieldValue, infoFieldValues));
+		}
+	}
+
+	private Optional<InfoFieldValue> _getInfoFieldValue(
+		T t, DDMFormFieldValue ddmFormFieldValue) {
+
+		Value value = ddmFormFieldValue.getValue();
+
+		if (value == null) {
+			return Optional.empty();
+		}
+
+		return Optional.of(
+			new InfoFieldValue<>(
+				new InfoField(
+					TextInfoFieldType.INSTANCE,
+					InfoLocalizedValue.localize(
+						getClass(), ddmFormFieldValue.getName()),
+					ddmFormFieldValue.getName()),
+				InfoLocalizedValue.builder(
+				).defaultLocale(
+					value.getDefaultLocale()
+				).addValues(
+					value.getValues(
+					).entrySet(
+					).stream(
+					).collect(
+						Collectors.toMap(
+							Map.Entry::getKey,
+							entry -> _sanitizeDDMFormFieldValue(
+								t, ddmFormFieldValue, entry.getKey()))
+					)
+				).build()));
+	}
+
+	private List<InfoFieldValue> _getInfoFieldValues(
+		T t, DDMFormFieldValue ddmFormFieldValue) {
+
+		List<InfoFieldValue> infoFieldValues = new ArrayList<>();
+
+		_addDDMFormFieldValue(t, ddmFormFieldValue, infoFieldValues);
+
+		return infoFieldValues;
+	}
+
+	private Object _sanitizeDDMFormFieldValue(
+		T t, DDMFormFieldValue ddmFormFieldValue, Locale locale) {
+
+		Value value = ddmFormFieldValue.getValue();
+
+		String valueString = value.getString(locale);
+
+		try {
+			if (Objects.equals(ddmFormFieldValue.getType(), "date") ||
+				Objects.equals(ddmFormFieldValue.getType(), "ddm-date")) {
+
+				DateFormat dateFormat = DateFormat.getDateInstance(
+					DateFormat.SHORT, locale);
+
+				Date date = DateUtil.parseDate(
+					"yyyy-MM-dd", valueString, locale);
+
+				return dateFormat.format(date);
+			}
+			else if (Objects.equals(
+						ddmFormFieldValue.getType(), "ddm-decimal") ||
+					 Objects.equals(ddmFormFieldValue.getType(), "numeric")) {
+
+				NumberFormat numberFormat = NumberFormat.getNumberInstance(
+					locale);
+
+				return numberFormat.format(GetterUtil.getDouble(valueString));
+			}
+			else if (Objects.equals(ddmFormFieldValue.getType(), "ddm-image") ||
+					 Objects.equals(ddmFormFieldValue.getType(), "image")) {
+
+				JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+					valueString);
+
+				jsonObject.put("url", _transformFileEntryURL(valueString));
+
+				return jsonObject;
+			}
+
+			return SanitizerUtil.sanitize(
+				t.getCompanyId(), t.getGroupId(), t.getUserId(),
```

Comment:
This sanitization seems to be the only reason why t is passed around, which makes the whole class uglier. However sanitization should not really be done here, since this class is not necessarily UI related. So a good solution to this would be to move sanitization somewhere else.

Reply:
Agreed.